src/server/project.ts

```diff
@@ -370,6 +410,54 @@ namespace ts.server {
             }
         }
 
+        getReferencedFiles(path: Path): Path[] {
+            if (!this.languageServiceEnabled) {
+                return [];
+            }
+
+            const sourceFile = this.getSourceFile(path);
+            if (!sourceFile) {
+                return [];
+            }
+            // We need to use a set here since the code can contain the same import twice,
+            // but that will only be one dependency.
+            // To avoid invernal conversion, the key of the referencedFiles map must be of type Path
+            const referencedFiles = createMap<boolean>();
+            if (sourceFile.imports) {
+                const checker: TypeChecker = this.program.getTypeChecker();
+                for (const importName of sourceFile.imports) {
+                    const symbol = checker.getSymbolAtLocation(importName);
+                    if (symbol && symbol.declarations && symbol.declarations[0]) {
+                        const declarationSourceFile = symbol.declarations[0].getSourceFile();
+                        if (declarationSourceFile) {
+                            referencedFiles[declarationSourceFile.path] = true;
+                        }
+                    }
+                }
+            }
+
+            const currentDirectory = getDirectoryPath(path);
+            const getCanonicalFileName = createGetCanonicalFileName(this.projectService.host.useCaseSensitiveFileNames);
+            // Handle triple slash references
+            if (sourceFile.referencedFiles) {
+                for (const referencedFile of sourceFile.referencedFiles) {
+                    const referencedPath = toPath(referencedFile.fileName, currentDirectory, getCanonicalFileName);
+                    referencedFiles[referencedPath] = true;
+                }
+            }
+
+            // Handle type reference directives
+            if (sourceFile.resolvedTypeReferenceDirectiveNames) {
+                for (const typeName in sourceFile.resolvedTypeReferenceDirectiveNames) {
+                    const fileName = sourceFile.resolvedTypeReferenceDirectiveNames[typeName].resolvedFileName;

```

Comment:
don't remember off the top of my head what will be stored in `resolvedTypeReferenceDirectiveNames` if `/// <reference types="blah"/>` cannot be resolved - probably `undefined`, similar to resolved module names. Can you check and add a guard against this case?